packages/training/python/src/training/schemas.py

```
@@ -0,0 +1,479 @@
+    # Validate numeric fields
+    pnl = data.get("finalPnL", data.get("final_pnl"))
+    if pnl is not None:
+        try:
+            float(pnl)
+        except (TypeError, ValueError):
+            errors.append(f"finalPnL must be a number, got {type(pnl).__name__}")
+    
+    episode_length = data.get("episodeLength", data.get("episode_length"))
+    if episode_length is not None:
+        if not isinstance(episode_length, int) or episode_length < 0:
+            errors.append(f"episodeLength must be a non-negative integer")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)